engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorImpl.java

```diff
@@ -259,11 +259,6 @@ public ValidatorImpl(ConstraintValidatorFactory constraintValidatorFactory,
 	}
 
 	private <T> Set<ConstraintViolation<T>> validateParameters(T object, Executable executable, Object[] parameterValues, Class<?>... groups) {
-		//this might be the case for parameterless methods
-		if ( parameterValues == null ) {
-			return Collections.emptySet();
-		}
-
```

Comment:
This removed if statement seems to be redundant - this private method is called from two places. And in both cases there's 
```
Contracts.assertNotNull( parameterValues, MESSAGES.validatedParameterArrayMustNotBeNull() );
```
So `null` cannot be passed to this method.